res/controllers/Reloop-Beatmix-2-4-scripts.js

```diff
@@ -0,0 +1,695 @@
+    for (var i = 0x91; i <= 0x94; i++) {
+        midi.sendShortMsg(i, 0x26 - mode, OFF);
+        midi.sendShortMsg(i, 0x26 + SHIFT - mode, OFF);
+    }
+    engine.beginTimer(150, "ReloopBeatmix24.FxModeLedFlash(1, " + mode + ")",
+        true);
+};
+
+// This function activate Fx Unit 1 or 2 for the selected Channel.
+// It is mapped to SHIFT + PITCHBEND+/- (FX1 and FX2)
+ReloopBeatmix24.ActivateFx = function(channel, control, value, status, group) {
+    // Calculate Fx num based on midi control (0x66 for Fx1 and 0x67 for Fx2)
+    var FxNum = control - 0x65;
+    if (value === DOWN) {
+        if (FxModeTimers[group]) {
+            engine.stopTimer(FxModeTimers[group]);
+            delete FxModeTimers[group];
+        }
+        FxModeLongPressed[group] = false;
+        FxModeTimers[group] = engine.beginTimer(1000,
+            "ReloopBeatmix24.FxModeCallback(\"" + group + "\", " + FxNum + ")",
+            true);
+    } else { // UP
+        if (FxModeLongPressed[group]) { // long press
+            // Nothing to do, this has already been done in callback function
+            FxModeLongPressed[group] = false;
+        } else { // short press
+            // stop & delete timer
+            engine.stopTimer(FxModeTimers[group]);
+            delete FxModeTimers[group];
+            script.toggleControl("[EffectRack1_EffectUnit" + FxNum + "]",
+                "group_" + group + "_enable");
+        }
+    }
+};
+
+ReloopBeatmix24.FxKnobTurn = function(channel, control, value, status, group) {
+    if (FxMode == 1) {
+        var parameter = control;
+        engine.setParameter(group, "parameter" + parameter.toString(),
+            script.absoluteLin(value, 0, 1));
+    }
+    // Nothing in multi-effect mode
+};
+
+ReloopBeatmix24.ShiftFxKnobTurn = function(channel, control, value, status,
+    group) {
+    if (FxMode == 1) {
+        var parameter = 3 + control - SHIFT;
+        engine.setParameter(group, "parameter" + parameter.toString(),
+            script.absoluteLin(value, 0, 1));
+    } else {
+        var effectUnit = parseInt(group.substr(23, 1), 10);
+        var Effect = control - SHIFT;
+        var storeIndex = "FX" + Effect.toString() + "U" + effectUnit.toString();
+        if (storeIndex in previousValue) {
+            if (value - previousValue[storeIndex] > 5) {
+                engine.setValue("[EffectRack1_EffectUnit" + effectUnit +
+                    "_Effect" + Effect + "]", "next_effect", 1);
+                previousValue[storeIndex] = value;
+            } else if (value - previousValue[storeIndex] < -5) {
+                engine.setValue("[EffectRack1_EffectUnit" + effectUnit +
+                    "_Effect" + Effect + "]", "prev_effect", 1);
+                previousValue[storeIndex] = value;
+            }
+        } else {
+            previousValue[storeIndex] = value;
+        }
+    }
+};
+
+ReloopBeatmix24.FxOff = function(channel, control, value, status, group) {
```

Comment:
This is undocumented behaviour, official manufacturer MIDI documentation doesn't talk about this. Another undocumented behaviour is when you push one of the "Change Deck" button, the controller sends an Identity Reply Universal Sysex Message showing the manufacturer of the MIDI interface chip, the family code, model number and version of the controller firmware.
Now, I know it's built by www.yahorng.com on behalt of Reloop ;)